### DISCUSSION TITLE
pixijs.io examples assets, displacement filters

### DISCUSSION THREAD

**Discussion by ripmurdock** [pixijs/pixijs#9159] [General]
Hi,

Are the assets used in the pixijs.io examples available for download?

In particular I'm interested in learning how to implement displacement filters:
https://pixijs.io/examples/#/filters-basic/displacement-map-flag.js

Thanks!

**Comment by bigtimebuddy** [comment]
https://pixijs.io/examples/examples/assets/pixi-filters/displacement_map_repeat.jpg

That's the image you're looking for. These assets are also checked into the repo: github.com/pixijs/examples

**Comment by ripmurdock** [comment]
Thanks!

This displacement filter flag could solve something I've been working on for days.

I wasn't able to get the example to work in codepen.  I'm getting no errors and this message in the console, so I believe the CDN reference is working: "PixiJS 6.5.4 - WebGL 2 - http://www.pixijs.com/"

I added a canvas element and tried replacing this line:
//document.body.appendChild(app.view);

with both of these lines:
document.querySelector("#pixi_container").appendChild(app.view);
document.getElementById("pixi_container").appendChild(app.view);

The PNG and JPG files both populate img elements with the links I used.

Any idea what I'm missing?

Thanks again.

**Reply by SuperSodaSea** [reply]
Could you provide the link to your CodePen?

**Comment by ripmurdock** [comment]
Thanks!

Here you go: https://codepen.io/ripmurdock/pen/eYLOBvX

I plan on using this animation on a Wordpress site using the Divi theme, so I'll need to figure out how to add the PIXI Application by selecting an element by id.

**Reply by SuperSodaSea** [reply]
Maybe you need one of these:
- Use `<div id="pixi_container"></div>` instead of `<canvas>`
- Or keep `<canvas>` and use `new PIXI.Application({ view: document.querySelector("#pixi_container") })`

BTW it seems that there are some errors shown in console caused by CORS:

```
Access to image at 'https://casadeveronica.com/wp-content/uploads/2023/02/basketclip.png' from origin 'https://cdpn.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
Access to image at 'https://casadeveronica.com/wp-content/uploads/2023/02/displacement_map_repeat.jpg' from origin 'https://cdpn.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by ripmurdock** [comment]
Thanks!

I also get a black square when selecting a <div>.  I do see those errors in the codepen Debug View, although the images populate in the codepen editor.  

Any suggestions where I can host images for testing pixijs in codepen?

I added the min js file to the site I'm working on, and the filter appears to be working:
https://casadeveronica.com/__2kdtkpy-z6huasdfasd_pixijs/

I do get an error message on this page related to the min js file:
DevTools failed to load source map: Could not load content for https://casadeveronica.com/wp-content/themes/divi-child/js/pixi.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

This appears to reference this text at the end of the file:
sourceMappingURL=pixi.min.js.map

Any idea if I made a mistake creating the pixijs.downloadv7.1.2pixi.min.js file from the screen scrape? 

It's a relief to see the pixijs filter working on the wordpress site.  I still hope to do some pixijs testing in codepen.

Thanks again!
